homeassistant/helpers/entityfilter.py

```diff
@@ -0,0 +1,74 @@
+"""Helper class to implement include/exclude of entities and domains."""
+
+from homeassistant.const import (
+    CONF_ENTITIES, CONF_DOMAINS)
+from homeassistant.core import split_entity_id
+
+
+class EntityFilter():
+    """Class that implements an include/exclude filter for entities/domains."""
+
+    def __init__(self, include_dict, exclude_dict):
+        """Set up the filter."""
+        self.include_e = include_dict.get(CONF_ENTITIES, [])
+        self.include_d = include_dict.get(CONF_DOMAINS, [])
+        self.exclude_e = exclude_dict.get(CONF_ENTITIES, [])
+        self.exclude_d = exclude_dict.get(CONF_DOMAINS, [])
+
+        if self.exclude_e + self.exclude_d == []:
+            self.have_exclude = False
+        else:
+            self.have_exclude = True
+
+        if self.include_e + self.include_d == []:
+            self.have_include = False
+        else:
+            self.have_include = True
+
+    def check_entity(self, entity_id):
+        """Check if a given entity_id should be filtered."""
+        domain = split_entity_id(entity_id)[0]
+
+        # Case 1 - no includes or excludes - pass all entities
+        if not self.have_include and not self.have_exclude:
+            return True
+
+        # Case 2 - includes, no excludes - only include specified entities
+        if self.have_include and not self.have_exclude:
+            return bool(entity_id in self.include_e or
```

Comment:
`in` checks already return booleans. The result of `or` is also a boolean. So no need to cast it to boolean.